test/CMakeLists.txt

```diff
@@ -32,12 +32,13 @@ set(gtest_hide_internal_symbols ON CACHE INTERNAL "Override gtest default")
 
 set(BUILD_GMOCK OFF CACHE INTERNAL "Override gtest default" FORCE)
 set(INSTALL_GTEST OFF CACHE INTERNAL "Override gtest default" FORCE)
-add_subdirectory(gtest EXCLUDE_FROM_ALL)
 
-if(CMAKE_COMPILER_IS_GNUCXX)
-    # Squash an error in gtest that surfaced in GCC 12
-    target_compile_options(gtest PRIVATE -Wno-restrict)
-endif()
+include(FetchContent)
+FetchContent_Declare(gtest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG v1.15.2
```

Comment:
I seem to recall 1.14.x caused issues with one of the older compilers we were trying to support, which is why we were previously locked on 1.13.0.

Happy to use the latest, but just want to confirm that you're intentionally bumping the version up.

Reply:
The submodule was actually on 1.12.1, which is the last version to support C++11 (newer versions require C++14).  I guess this technically means we'd be dropping support for C++11 for testing, but I think I'm fine with that.